File: src/main/java/org/dukcode/ps/codetree/trail02/chapter02/lesson02/ChallengeSequenceOfRemainderDividedBy100.java
package org.dukcode.ps.codetree.trail02.chapter02.lesson02;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.StringTokenizer;

/**
 * @see <a
 * href="https://www.codetree.ai/ko/trails/complete/curated-cards/challenge-sequence-of-remainder-divided-by-100/description">LINK</a>
 */
public class ChallengeSequenceOfRemainderDividedBy100 {

  private static BufferedReader br;
  private static BufferedWriter bw;
  private static StringTokenizer st;

  public static void main(String[] args) throws IOException {
    br = new BufferedReader(new InputStreamReader(System.in));
    bw = new BufferedWriter(new OutputStreamWriter(System.out));

    int n = Integer.parseInt(br.readLine());

    bw.write(String.valueOf(solve(n)));

    br.close();
    bw.close();
  }

  private static int solve(int n) {
    if (n == 1) {
      return 2;
    }

    if (n == 2) {
      return 4;
    }

    return (solve(n - 1) * solve(n - 2)) % 100;
  }
}

File: src/main/kotlin/org/dukcode/ps/codetree/trail02/chapter02/lesson02/ChallengeSequenceOfRemainderDividedBy100.kt
package org.dukcode.ps.codetree.trail02.chapter02.lesson02.challengeSequenceOfRemainderDividedBy100

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.StringTokenizer

private val br = BufferedReader(InputStreamReader(System.`in`))
private val bw = BufferedWriter(OutputStreamWriter(System.out))

/**
 * @see <a href="https://www.codetree.ai/ko/trails/complete/curated-cards/challenge-sequence-of-remainder-divided-by-100/description">LINK</a>
 */
fun main() {
    val n = br.readLine().toInt()

    bw.write(seq(n).toString())

    bw.flush()
}

private fun seq(n: Int): Int {
    if (n <= 2) {
        return n * 2
    }

    return seq(n - 1) * seq(n - 2) % 100
}

private fun parseInts() =
    StringTokenizer(br.readLine()).run {
        IntArray(countTokens()) {
            nextToken().toInt()
        }
    }
